### ISSUE TITLE
Crash when adding effects in Xcode 8.3

### ISSUE THREAD

**Issue by markjeschke** [audiokit/AudioKit#846]
Dear Aure and team,

In Xcode 8.3, when I attempt to add `AKDelay` or `AKReverb` effects nodes into the chain, my app crashes. Here's my code:
```
// The app crashes when I add delay and/or reverb effects. This same code worked prior to Xcode 8.3.
// Set enableEffects to true to test it.
var enableEffects: Bool = false

init() {
        
        do {
            try self.kickSampler.loadWav("Sounds/min_kick_02_C")
            try self.snareSampler.loadWav("Sounds/Ensoniq-ESQ-1-Snare")
        } catch {
            print("Could not locate the wav files.")
        }
        
        // Combine the samples with the oscillator into a mixer.
        let mixer = AKMixer(kickSampler, snareSampler, oscillator)
         
        if enableEffects {
            // Add a delay effect node to its output.
            let delay = AKDelay(mixer)
            
            delay.time = 1.0 // seconds
            delay.feedback  = 0.1 // Normalized Value 0 - 1
            delay.dryWetMix = 0.0  // Normalized Value 0 - 1
            delay.lowPassCutoff = 15000
            
            // Add a reverb effect after the delay node.
            let reverb = AKReverb(delay)
            
            reverb.dryWetMix = 0.2
            reverb.loadFactoryPreset(.largeRoom)
            
            // Pass the end of the audio chain to the global output.
            AudioKit.output = reverb
            print("Added delay and reverb node effects")
        } else {
            AudioKit.output = mixer
            print("Didn't include delay and reverb node effects. No crash.")
        }
        
        // Start the audio engine
        AudioKit.start()
        
        print("Audio engine started")
}
```
Here's my [AudioTest](https://github.com/markjeschke/Audio-Test) repo, if you're interested in testing and troubleshooting my code. Any help would be greatly appreciated!

Thanks,
Mark

TL;DR: 

Here's the console log output error that I receive upon the crash:
```
[central] 54:   ERROR:    >avae> AVAudioEngineGraph.mm:1365: RemoveNode: required condition is false: !nodeMixerConns.empty() && !hasDirectConnToIONode
2017-04-10 00:52:38.585 Audio Test[37450:746637] *** Terminating app due to uncaught exception 'com.apple.coreaudio.avfaudio', reason: 'required condition is false: !nodeMixerConns.empty() && !hasDirectConnToIONode'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010d9a9b0b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010a8b2141 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010d9adcf2 +[NSException raise:format:arguments:] + 98
	3   AVFAudio                            0x000000010ffeea9e _Z19AVAE_RaiseExceptionP8NSStringz + 158
	4   AVFAudio                            0x000000010ff96e79 _ZN18AVAudioEngineGraph10RemoveNodeEP11AVAudioNode + 1019
	5   AVFAudio                            0x000000010fff75aa _ZN17AVAudioEngineImpl16DestroyGraphNodeEP11AVAudioNode + 18
	6   AVFAudio                            0x000000010ffe8829 -[AVAudioNode didDetachFromEngine:] + 77
	7   AVFAudio                            0x000000010fff4cd1 _ZN17AVAudioEngineImpl10DetachNodeEP11AVAudioNodeb + 443
	8   AVFAudio                            0x000000010fff4ad0 -[AVAudioEngine detachNode:] + 67
	9   AudioKit                            0x0000000109d05f94 _TToFC8AudioKit6AKNodeD + 84
	10  libobjc.A.dylib                     0x000000010a8c6b8e _ZN11objc_object17sidetable_releaseEb + 202
	11  Audio Test                          0x0000000109aab3ef _TFC10Audio_Test11AudioEnginecfT_S0_ + 1711
	12  Audio Test                          0x0000000109aaac9f _TFC10Audio_Test11AudioEngineCfT_S0_ + 47
	13  Audio Test                          0x0000000109aaac61 globalinit_33_88E80B9B2E3B800627B74484753981AD_func0 + 17
	14  libdispatch.dylib                   0x000000010f9a205c _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x000000010f9879a1 dispatch_once_f + 503
	16  Audio Test                          0x0000000109aaad28 _TFC10Audio_Test11AudioEngineau14sharedInstanceS0_ + 40
	17  Audio Test                          0x0000000109aac6ad _TFC10Audio_Test14ViewControllercfT5coderCSo7NSCoder_GSqS0__ + 29
	18  Audio Test                          0x0000000109aac79d _TToFC10Audio_Test14ViewControllercfT5coderCSo7NSCoder_GSqS0__ + 45
	19  UIKit                               0x000000010b18bf54 -[UIClassSwapper initWithCoder:] + 246
	20  UIKit                               0x000000010b37c4af UINibDecoderDecodeObjectForValue + 693
	21  UIKit                               0x000000010b37c1ef -[UINibDecoder decodeObjectForKey:] + 89
	22  UIKit                               0x000000010b18bc20 -[UIRuntimeConnection initWithCoder:] + 178
	23  UIKit                               0x000000010b37c4af UINibDecoderDecodeObjectForValue + 693
	24  UIKit                               0x000000010b37c64d UINibDecoderDecodeObjectForValue + 1107
	25  UIKit                               0x000000010b37c1ef -[UINibDecoder decodeObjectForKey:] + 89
	26  UIKit                               0x000000010b18ae1d -[UINib instantiateWithOwner:options:] + 1262
	27  UIKit                               0x000000010b54382f -[UIStoryboard instantiateViewControllerWithIdentifier:] + 181
	28  UIKit                               0x000000010ad838d1 -[UIApplication _loadMainStoryboardFileNamed:bundle:] + 108
	29  UIKit                               0x000000010ad83db3 -[UIApplication _loadMainInterfaceFile] + 260
	30  UIKit                               0x000000010ad82456 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1407
	31  UIKit                               0x000000010ad7f793 -[UIApplication workspaceDidEndTransaction:] + 182
	32  FrontBoardServices                  0x000000011146a5f6 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 24
	33  FrontBoardServices                  0x000000011146a46d -[FBSSerialQueue _performNext] + 186
	34  FrontBoardServices                  0x000000011146a7f6 -[FBSSerialQueue _performNextFromRunLoopSource] + 45
	35  CoreFoundation                      0x000000010d94fc01 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	36  CoreFoundation                      0x000000010d9350cf __CFRunLoopDoSources0 + 527
	37  CoreFoundation                      0x000000010d9345ff __CFRunLoopRun + 911
	38  CoreFoundation                      0x000000010d934016 CFRunLoopRunSpecific + 406
	39  UIKit                               0x000000010ad7e02f -[UIApplication _run] + 468
	40  UIKit                               0x000000010ad840d4 UIApplicationMain + 159
	41  Audio Test                          0x0000000109aad3b7 main + 55
	42  libdyld.dylib                       0x000000010f9ee65d start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

**Comment by aure** [comment]
I told this to you in Slack, the mixer, reverb, and delay effects are local and not retained, they need to be instance variables.
